juju/juju
# feat(instancedata): remove HardwareCharacteristics method from state/machine

This patch is part of the instance data removal from the legacy state.

Usages of `HardwareCharacteristics()` not yet wired are replaced by the
machine domain method.





## Checklist




## QA steps

Bootstrap, add a machine and deploy ubuntu to lxd:
```
juju bootstrap aws c
juju add-model m
juju add-machine
juju deploy ubuntu --to lxd
```
Then you can make sure the status is correct, make sure to see the hardware:
```
juju status --format yaml | grep "hardware"
    hardware: arch=amd64 cores=2 cpu-power=895 mem=8192M root-disk=8192M availability-zone=eu-west-3a
    hardware: arch=amd64 cores=2 cpu-power=895 mem=8192M root-disk=8192M availability-zone=eu-west-3b
```

**UPDATED: Bootstrap on aws.**

## Links


**Jira card:** JUJU-6703
